intelmq/bin/intelmqsetup.py

```diff
@@ -17,59 +22,134 @@
+MANAGER_CONFIG_DIR = Path(CONFIG_DIR) / 'manager/'
+FILE_OUTPUT_PATH = Path(VAR_STATE_PATH) / 'file-output/'
 
+
+def basic_checks(skip_ownership):
+    if os.geteuid() != 0 and not skip_ownership:
+        sys.exit(red('You need to run this program as root for setting file ownership!'))
     if not ROOT_DIR:
-        sys.exit('Not a pip-installation of IntelMQ, nothing to initialize.')
-
-    create_dirs = ('%s/file-output' % VAR_STATE_PATH,
-                   VAR_RUN_PATH,
-                   DEFAULT_LOGGING_PATH,
-                   CONFIG_DIR)
-    for create_dir in create_dirs:
-        if not os.path.isdir(create_dir):
-            os.makedirs(create_dir, mode=0o755,
-                        exist_ok=True)
-            print('Created directory %r.' % create_dir)
+        sys.exit(red('Not a pip-installation of IntelMQ, nothing to initialize.'))
+
+    if skip_ownership:
+        return
+    try:
+        getpwnam('intelmq')
+    except KeyError:
+        sys.exit(red("User 'intelmq' does not exist. Please create it and then re-run this program."))
+    try:
+        getgrnam('intelmq')
+    except KeyError:
+        sys.exit(red("Group 'intelmq' does not exist. Please create it and then re-run this program."))
+
+
+def create_directory(directory: str, octal_mode: int):
+    readable_mode = stat.filemode(octal_mode)
+    if not Path(directory).is_dir():
+        Path(directory).mkdir(mode=octal_mode, exist_ok=True, parents=True)
+        print(f'Created directory {directory!s} with permissions {readable_mode}.')
+    else:
+        current_mode = Path(directory).stat().st_mode
+        if current_mode != octal_mode:
+            current_mode_readable = stat.filemode(current_mode)
+            print(f'Fixed wrong permissions of {directory!s}: {current_mode_readable!r} -> {readable_mode!r}.')
+            Path(directory).chmod(octal_mode)
+
+
+def change_owner(file: str, owner=None, group=None):
+    if owner and Path(file).owner() != owner:
+        print(f'Fixing owner of directory {file!r}.')
+        shutil.chown(file, user=owner)
+    if group and Path(file).group() != group:
+        print(f'Fixing group of directory {file!r}.')
+        shutil.chown(file, group=group)
+
+
+def find_webserver_user():
+    candidates = ('www-data', 'wwwrun', 'httpd', 'apache')
+    for candidate in candidates:
+        try:
+            getpwnam(candidate)
+        except KeyError:
+            pass
+        else:
+            print(f'Detected webserver username {candidate!r}.')
+            return candidate
+
+
+def intelmqsetup_core(ownership=True, state_file=STATE_FILE_PATH):
+    create_directory(FILE_OUTPUT_PATH, 0o40755)
+    create_directory(VAR_RUN_PATH, 0o40755)
+    create_directory(DEFAULT_LOGGING_PATH, 0o40755)
+    create_directory(CONFIG_DIR, 0o40775)
 
     example_confs = glob.glob(pkg_resources.resource_filename('intelmq', 'etc/*.conf'))
     for example_conf in example_confs:
```

Comment:
You could also use the `glob` method of the pathlib module here:
```
pathlib.Path(pkg_resources.resource_filename('intelmq', 'etc')).glob('*.conf')
```
which gives you `PosixPath` objects to work with- then you don't have to use `Path()` around the name.